src/main/com/intellij/lang/jsgraphql/ide/project/GraphQLUIProjectService.java

```diff
@@ -311,7 +313,17 @@ public void executeGraphQL(Editor editor, VirtualFile virtualFile) {
                 final GraphQLQueryContext context = GraphQLQueryContextHighlightVisitor.getQueryContextBufferAndHighlightUnused(editor);
 
                 Map<String, Object> requestData = new HashMap<>();
-                requestData.put("query", context.query);
+                Pattern pattern = Pattern.compile("(query|mutation|subscription) (.*) ", Pattern.CASE_INSENSITIVE);
```

Comment:
basically this allows regex to find the name of query to be added to operationName key in data object, if no name is detected, "anonymous" is added to query and operationName key as a workaround since our graphql server wouldn't allow truly nameless queries